app/vmselect/promql/exec.go

```diff
@@ -154,7 +155,16 @@ func removeNaNs(tss []*timeseries) []*timeseries {
 	return rvs
 }
 
-func parsePromQLWithCache(q string) (expr, error) {
+func parsePromQL(q string) (promql.Expr, error) {
+	e, err := parser.ParsePromQL(q)
```

Comment:
Note that `parser.ParsePromQL` differs slightly from what `parsePromQL` previously did: in the past, the result of parsing a query had the `WITH` clause expanded, and numerical constants simplified. Since simplification needs all the evaluation functions, `parser.ParsePromQL` does not do simplification, and this method applies for use in this package.